includes/reader-activation/class-reader-activation.php

```diff
@@ -1699,6 +1690,41 @@ public static function register_reader( $email, $display_name = '', $authenticat
 		return $user_id;
 	}
 
+	/**
+	 * Get user data args for creating a new WP user.
+	 * See https://developer.wordpress.org/reference/functions/wp_insert_user/ for supported args.
+	 *
+	 * @param array $user_data          Default args for the new user.
+	 *              $user_data['email] Email address for the new user (required).
+	 */
+	public static function get_user_data( $user_data = [] ) {
+		if ( empty( $user_data['user_email'] ) ) {
+			return $user_data;
+		}
+
+		// If we don't have a display name, make it match the email address.
```

Comment:
When making a donation via the checkout modal, the `$user_data' passed in the call to this filter does not include `display_name`, which results in email address being used as the base for `display_name`. For this reason, the second testing scenario cannot be fulfilled.

Reply:
It's true that the [`woocommerce_new_customer_data` filter](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/wc-user-functions.php#L96-L107) doesn't include `display_name`. However, if a new customer account is created during checkout, the checkout method to do so [adds the display name based on billing first/last name](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-checkout.php#L1161) if available. So the effect is the same.

This condition is actually here for the benefit of [our own `register_reader` method](https://github.com/Automattic/newspack-plugin/pull/2845/files#diff-7f238c1a510d794797749876c3ac59626a1672f70f67e0dc41873ac9337f9793R1623-R1628), which calls `wc_create_new_customer` outside of the checkout. Our method is called to register readers via the Register and Newsletter Subscription Form blocks, and the latter includes optional name fields which might be passed and used to build the display name if available.